integreat_cms/summ_ai_api/summ_ai_api_client.py

```diff
@@ -122,27 +136,44 @@ async def translate_text_fields(self, loop, text_fields):
         :returns: The list of completed text fields
         :rtype: list [ ~integreat_cms.summ_ai_api.utils.TextField ]
         """
+
         # Set a custom SUMM.AI timeout
         timeout = aiohttp.ClientTimeout(total=60 * settings.SUMM_AI_TIMEOUT)
         async with aiohttp.ClientSession(timeout=timeout) as session:
             # Create tasks for each text field
             tasks = [
-                loop.create_task(self.translate_text_field(session, text_field))
+                # translate_text_field() gives us a coroutine that can be executed
+                # asyncronously as a task. If we have to repeat the task
+                # (e.g. if we run into rate limiting and have to resend the request),
+                # we need a NEW coroutine object.
+                # For that case, we a representation of our function which can be
+                # evaluated when needed, giving a new coroutine for the task each time.
+                delay_function_call(self.translate_text_field, [session, text_field])
                 for text_field in text_fields
             ]
+            # A "patient" task queue which only hands out sleep tasks after a task was reported as failed
+            task_generator = PatientTaskQueue(tasks)
+
             # Wait for all tasks to finish and collect the results
-            # (the results are sorted in the order the tasks were created)
-            return await asyncio.gather(*tasks)
+            worker_results = await asyncio.gather(
+                *[
+                    worker(loop, task_generator, str(i))
```

Comment:
Yeah, but it won't be any significant overhead to keep in case we want to instantiate different groups of workers and distinguish their log entries down the road.

Reply:
In logging, the `%s` formatter already converts ints to strings, so you wouldn't have to change the logging calls.